src/core/uri/uri.cc

```diff
@@ -139,16 +139,11 @@ URI::~URI() { uriFreeUriMembersA(&this->internal->uri); }
 URI::URI(const URI &other) : URI{other.recompose()} {}
 
 URI::URI(URI &&other)
-    : data{std::move(other.data)}, internal{std::move(other.internal)} {
-  this->parsed = other.parsed;
-  this->path_ = std::move(other.path_);
-  this->scheme_ = std::move(other.scheme_);
-  this->userinfo_ = std::move(other.userinfo_);
-  this->host_ = std::move(other.host_);
-  this->port_ = other.port_;
-  this->fragment_ = std::move(other.fragment_);
-  this->query_ = std::move(other.query_);
-
+    : parsed{other.parsed}, data{std::move(other.data)},
```

Comment:
This is not about the check but made the change from advantage I understood about constructor initialiser list.

Reply:
Makes sense! Good catch